### DISCUSSION TITLE
XAML previewer on mac (VS)

### DISCUSSION THREAD

**Discussion by mijenner** [AvaloniaUI/Avalonia#7932] [Q&A]
I'm planning on using Avalonia for high school programming classes. We have chosen C#, Visual Studio Community, that we want to go a little beyond the console and are facing students with a mix of Windows and Mac computers. This turned out not to be the easiest route to take (python might have been easier but ...). 

We've used Unity which is truly cross platform but loaded with a lot of "extras". We've considered Uno platform which has some potential but also missing a lot elements on different platforms. I was happy to discover the articles on Avalonia by Nick Polyak on codeproject - now I think this is a useful route for us to take. 

Conclusion so far: Avalonia seems like the most mature cross platform UI around, great work! 

However, now I'm facing problems getting the XAML previewer to run on Visual Studio on mac. the [AvaloniaVS instructions](https://github.com/AvaloniaUI/AvaloniaVS) seem to target Windows and not mention mac at all. Anyone with success getting this to run on mac that would share some hints? 

Kind regards, 
Michael

**Comment by wieslawsoltes** [comment]
I would recommend using JetBrains Rider as it has great Avalonia xaml previewer working on Windows, macOS and Linux. Rider is not free, but maybe they have license for education.

https://docs.avaloniaui.net/docs/getting-started/ide-support/jetbrains-rider-setup

**Comment by grokys** [comment]
To follow on from @wieslawsoltes : the problem is that Visual Studio for Mac is a completely different product than Visual Studio for Windows, and has a completely different extension API, so we'd have to write a completely new extension for that product, which we don't currently have the manpower for.

**Comment by MarkusKgit** [comment]
I don't have a Mac but you could try to build [AvantGarde](https://github.com/kuiperzone/AvantGarde)

It's not ideal as you would have to open VS + this but maybe it works for you.
